lib/matplotlib/tests/test_font_manager.py

```diff
@@ -287,6 +287,28 @@ def test_fontcache_thread_safe():
     subprocess_run_helper(_test_threading, timeout=10)
 
 
+def test_lockfilefailure(tmp_path):
+    # The logic here:
+    # 1. get a temp directory from pytest
+    # 2. import matplotlib which makes sure it exists
+    # 3. get the cache dir (where we check it is writable)
+    # 4. make it not writable
+    # 5. try to write into it via font manager
```

Comment:
```suggestion
    # 5. try to write into it via font manager
    # 6. Check that we do not raise even though 5. cannot work
```

Reply:
Optional: Check that the log warning is emitted.

Reply:
I'm going to pass on adding a check for the logging.  I don't think we do that systematically currently and because the log emission happens in the sub-process it would take a fair amount of work to check (I think something like the tests for blocking where we print something and them search for the string in stdout).